### PR TITLE
Update istio to version 1.23.2

### PR DESCRIPTION
**Pull request by axel7born**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Update istio to version 1.23.2
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -56,11 +56,11 @@ require (
 	google.golang.org/protobuf v1.34.2
 	gopkg.in/yaml.v3 v3.0.1
 	helm.sh/helm/v3 v3.14.4
-	istio.io/api v1.22.5
-	istio.io/client-go v1.22.0
-	k8s.io/api v0.29.9
+	istio.io/api v1.23.1
+	istio.io/client-go v1.23.1
+	k8s.io/api v0.30.0
```
Comment (acumino) [inline go.mod]: K8s dependencies will be updated by - https://github.com/gardener/gardener/pull/10459
/hold

Comment (ScheererJ) [reply go.mod]: #10459 has been merged.

/unhold

@axel7born Please rebase.

**Review by ScheererJ — APPROVED** [review]
/lgtm
/approve

**Pull request merged**